xbmc/threads/Condition.h

```diff
@@ -20,14 +20,59 @@
 
 #pragma once
 
-#include "threads/platform/Condition.h"
-
+#include "threads/SingleLock.h"
+#include "threads/Helpers.h"
 #include "threads/SystemClock.h"
-#include <stdio.h>
+
+#include <condition_variable>
+#include <chrono>
 
 namespace XbmcThreads
 {
 
+  /**
+   * This is a thin wrapper around std::condition_variable_any. It is subject
+   *  to "spurious returns" as it is built on boost which is built on posix
+   *  on many of our platforms.
+   */
+  class ConditionVariable : public NonCopyable
```

Comment:
Can get rid of NonCopyable and just delete the copy constructors.
Not necessary but cleans things up a bit